scripts/skills/perks/perk_rf_tempo.nut

```diff
@@ -22,43 +24,46 @@ this.perk_rf_tempo <- ::inherit("scripts/skills/skill", {
 
 	function isHidden()
 	{
-		return this.m.Stacks == 0;
+		return this.m.Stacks == 0 && this.m.APRecovery.len() == 0;
 	}
 
 	function getTooltip()
 	{
 		local tooltip = this.skill.getTooltip();
 
-		tooltip.push({
-			id = 10,
-			type = "text",
-			icon = "ui/icons/initiative.png",
-			text = "[color=" + ::Const.UI.Color.PositiveValue + "]+" + this.getBonus() + "[/color] Initiative"
-		});
+		if (this.m.Stacks != 0)
+		{
+			tooltip.push({
+				id = 10,
+				type = "text",
+				icon = "ui/icons/initiative.png",
+				text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorizeValue(this.getBonus()) + " [Initiative|Concept.Initiative]")
+			});
+		}
+
+		if (this.m.APRecovery.len() != 0)
+		{
+			tooltip.push({
+				id = 10,
+				type = "text",
+				icon = "ui/icons/action_points.png",
+				text = ::Reforged.Mod.Tooltips.parseString("The next attack, if successful, against a target that acts after you in this [round|Concept.Round] will recover " + ::MSU.Text.colorGreen(this.m.APRecovery[0]) + " [Action Points|Concept.ActionPoints]")
+			});
+		}
 
 		if (this.m.IsPrimed)
 		{
 			tooltip.push({
 				id = 10,
 				type = "text",
 				icon = "ui/icons/warning.png",
-				text = ::MSU.Text.colorRed("This bonus has been carried over from the previous turn and will expire after using a skill or upon waiting or ending this turn")
+				text = ::MSU.Text.colorRed("The Initiative bonus has been carried over from the previous turn and will expire after using a skill or upon waiting or ending this turn")
```

Comment:
Any skill/attack that might scale off of initiative. The most relevant case is that of fencers, so that they can use their first lunge on the next turn with the built-up initiative. If they lose that initiative then the perk becomes significantly less useful for them.